presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -231,10 +231,8 @@ boolean internalExecute(String sql)
         clearCurrentResults();
         checkOpen();
 
-        StatementClient client = null;
         ResultSet resultSet = null;
-        try {
-            client = connection().startQuery(sql, getStatementSessionProperties());
+        try (StatementClient client = connection().startQuery(sql, getStatementSessionProperties())) {
```

Comment:
This is also incorrect. We assign the client to a field if the query is still running. We only close it if the query has finished.

Reply:
thanks for clarification. This is my mistake